### ISSUE TITLE
E: Package 'supervisor' has no installation candidate

### ISSUE THREAD

**Issue by xorpd** [docker/docker#16307]
Hi,
I can't manage to install supervisor on a simple Docker built over ubuntu:trusty. This is my Dockerfile:

    FROM ubuntu:trusty
    MAINTAINER xorpd <xorpd@xorpd.net>

    # Update packages:
    RUN apt-get update
    RUN apt-get upgrade -qqy

    # Get supervisor:
    RUN apt-get install -y supervisor

This is what I get when I try to build:

	$ docker build -no-cache -t fcatalog_server .
	Warning: '-no-cache' is deprecated, it will be replaced by '--no-cache' soon. See usage.
	Sending build context to Docker daemon  29.7 kB
	Step 0 : FROM ubuntu:trusty
	 ---> 91e54dfb1179
	Step 1 : MAINTAINER xorpd <xorpd@xorpd.net>
	 ---> Running in c7bc57793134
	 ---> 0be3d9a98b19
	Removing intermediate container c7bc57793134
	Step 2 : RUN apt-get update
	 ---> Running in 5f26f9200450
	Ign http://archive.ubuntu.com trusty InRelease
	Ign http://archive.ubuntu.com trusty-updates InRelease
	Ign http://archive.ubuntu.com trusty-security InRelease
	Get:1 http://archive.ubuntu.com trusty Release.gpg [72 B]
	Get:2 http://archive.ubuntu.com trusty-updates Release.gpg [933 B]
	Get:3 http://archive.ubuntu.com trusty-security Release.gpg [933 B]
	Get:4 http://archive.ubuntu.com trusty Release [11.9 kB]
	Get:5 http://archive.ubuntu.com trusty-updates Release [63.5 kB]
	Err http://archive.ubuntu.com trusty Release
	  
	Get:6 http://archive.ubuntu.com trusty-security Release [63.5 kB]
	Get:7 http://archive.ubuntu.com trusty-updates/main Sources [294 kB]
	Get:8 http://archive.ubuntu.com trusty-updates/restricted Sources [4521 B]
	Get:9 http://archive.ubuntu.com trusty-updates/universe Sources [170 kB]
	Get:10 http://archive.ubuntu.com trusty-updates/main amd64 Packages [779 kB]
	Get:11 http://archive.ubuntu.com trusty-updates/restricted amd64 Packages [22.7 kB]
	Get:12 http://archive.ubuntu.com trusty-updates/universe amd64 Packages [407 kB]
	Get:13 http://archive.ubuntu.com trusty-security/main Sources [118 kB]
	Get:14 http://archive.ubuntu.com trusty-security/restricted Sources [1874 B]
	Get:15 http://archive.ubuntu.com trusty-security/universe Sources [34.8 kB]
	Get:16 http://archive.ubuntu.com trusty-security/main amd64 Packages [432 kB]
	Get:17 http://archive.ubuntu.com trusty-security/restricted amd64 Packages [14.8 kB]
	Get:18 http://archive.ubuntu.com trusty-security/universe amd64 Packages [150 kB]
	Fetched 2568 kB in 11s (224 kB/s)
	Reading package lists...
	W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: http://archive.ubuntu.com trusty Release: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 16126D3A3E5C1192

	W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty/Release  

	W: Some index files failed to download. They have been ignored, or old ones used instead.
	 ---> 09486a44e22e
	Removing intermediate container 5f26f9200450
	Step 3 : RUN apt-get upgrade -qqy
	 ---> Running in 3f63bd27662a
	debconf: unable to initialize frontend: Dialog
	debconf: (TERM is not set, so the dialog frontend is not usable.)
	debconf: falling back to frontend: Readline
	debconf: unable to initialize frontend: Readline
	debconf: (This frontend requires a controlling tty.)
	debconf: falling back to frontend: Teletype
	dpkg-preconfigure: unable to re-open stdin: 
	(Reading database ... 11541 files and directories currently installed.)
	Preparing to unpack .../libexpat1_2.1.0-4ubuntu1.1_amd64.deb ...
	Unpacking libexpat1:amd64 (2.1.0-4ubuntu1.1) over (2.1.0-4ubuntu1) ...
	Preparing to unpack .../sudo_1.8.9p5-1ubuntu1.2_amd64.deb ...
	invoke-rc.d: policy-rc.d denied execution of stop.
	Unpacking sudo (1.8.9p5-1ubuntu1.2) over (1.8.9p5-1ubuntu1.1) ...
	Processing triggers for ureadahead (0.100.0-16) ...
	Setting up libexpat1:amd64 (2.1.0-4ubuntu1.1) ...
	Setting up sudo (1.8.9p5-1ubuntu1.2) ...
	Processing triggers for libc-bin (2.19-0ubuntu6.6) ...
	 ---> 1674e299f421
	Removing intermediate container 3f63bd27662a
	Step 4 : RUN apt-get install -y supervisor
	 ---> Running in 23f0209aade6
	Reading package lists...
	Building dependency tree...
	Reading state information...
	Package supervisor is not available, but is referred to by another package.
	This may mean that the package is missing, has been obsoleted, or
	is only available from another source

	E: Package 'supervisor' has no installation candidate
	The command '/bin/sh -c apt-get install -y supervisor' returned a non-zero code: 100

**Comment by cpuguy83** [comment]
apt-get update and apt-get install should always be done on a single line.
Closing as this is not really a docker issue.
For questions/support, please use either the docker-user mailing list, #docker on IRC, or forums.docker.com
Thanks!

**Comment by xorpd** [comment]
@cpuguy83: Thank you for your help. You saved me a lot of time.

I don't think that the statement "apt-get update and apt-get install should always be done on a single line" is general knowledge. Some of my old Dockerfiles from 6 months ago have separate apt-get update and apt-get install RUN commands and they used to work properly. I checked them again today and they are not working anymore.

A google search for Dockerfile examples gives many repositories where apt-get update and install are on separate lines. See for example: https://github.com/kstaken/dockerfile-examples

After you replied I did manage to find a note about it at the documentation here:
https://docs.docker.com/reference/builder/#run
